### PR TITLE
DUPLO-5965:  Add EphemeralStorage to lambda

### PR DESCRIPTION
**Pull request by michalpleszkun**

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-5965 Lambdas need the ability to have ephemeral storage: Add APIs to TF](https://app.clickup.com/t/8655600/DUPLO-5965)

File: duplocloud/resource_duplo_aws_lambda_function.go [hunk 1]
```diff
@@ -388,6 +398,7 @@ func flattenAwsLambdaConfiguration(d *schema.ResourceData, duplo *duplosdk.Duplo
 	d.Set("handler", duplo.Handler)
 	d.Set("version", duplo.Version)
 	d.Set("layers", duplo.Layers)
+	d.Set("ephemeral_storage", duplo.EphemeralStorage.Size)
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_aws_lambda_function.go:401]: `duplo.EphemeralStorage` can be `nil` but we aren't checking for it here.

**Review by joek-duplo — CHANGES_REQUESTED** [review]
we also forgot to set up ephemeral storage in the Update method

**Comment by ddiaz-duplo** [comment]
Could use some help testing this. I was using the example that @michalpleszkun added but I get the following error `Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist."`

```terraform
resource "duplocloud_s3_bucket" "s3bucket" {
  tenant_id = var.tenant_id
  name      = "miketest"
}

resource "duplocloud_aws_lambda_function" "test" {
  tenant_id         = var.tenant_id
  name              = "miketest"
  package_type      = "Zip"
  description       = "miketest lambda"
  s3_bucket         = duplocloud_s3_bucket.s3bucket.fullname
  s3_key            = "file.zip"
  memory_size       = 128
  runtime           = "python3.9"
  handler           = "main"
  ephemeral_storage = 1024
}
```
how do I retrieve that key?

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**